src/twin/mod.rs

```diff
@@ -334,6 +334,11 @@ impl Twin {
             None
         };
 
+        let mut update_validation = UpdateValidation::default();
+        // has to be called before iothub client authentication
+        update_validation.init()?;
```

Comment:
Is there a reason to split creation and initialization?

Reply:
personal preference to not have a constructor wich initializes depending on external state (if-else statement). 
jan asked the same ... if you both vote for it, i'll adapt.

Reply:
Just a matter of taste, what I like to do in such situations is using sth. like a static "create_for_...".

Reply:
for initialization I'm aware of `::new()`, `::default()` or `::from(some_args)`. 
from that point of view I would go for a `new()` here. a static create fn would also be fine.

Reply:
implemented `new()`